### DISCUSSION TITLE
Windows randomly deleted and scuffed my .kdbx database?

### DISCUSSION THREAD

**Discussion by yogurt231** [keepassxreboot/keepassxc#9742] [Q&A] [answered]
I didn't know where else to post this.

I have one SSD with Windows installed on it, and another one with Linux installed on it. I have one HDD (NTFS) where my .kdbx database is located, so that I can access it from both OSes. For the past week, I've only been booting to my Linux SSD. Today, I booted into Windows, navigated to my HDD to make sure my .kdbx database was there, and after a few seconds it just... disappeared in front of my eyes. I plugged in my backup USB where I have a backup of the .kdbx and Windows immediately said something like "Something is wrong with the USB, choose what to do" or whatever and I clicked "Open anyway". The .kdbx file on this USB then turns to 0 kB in size for some reason. I plug in my second backup USB and same thing happened.

Anyway, for the original .kdbx file that disappeared from my HDD, I downloaded the Windows File Recovery from Windows Store (winfr command), found the deleted .kdbx, and then restored it. Now I have the "restored" .kdbx, and it reports the correct file size, but when I try to unlock it in KeePassXC, it gives the "Not a KeePass database" error.

Is there any way I can make the .kdbx file accessible again? It happens both on Linux and Windows. All my drives and USB sticks are healthy and new. The .kdbx file and its backups worked fine this morning on Linux. On Windows, I use the Bitdefender Free antivirus. I can't see that it has deleted any files in its notifications log.

Thanks!

**Comment by phoerious** [comment, marked as answer]
I don't think we can help you with this and since it's not KeePassXC messing with your file, we can't even try to find the cause for your mishap. Restoring deleted files is extremely unreliable, especially for encrypted containers and even more so if you don't remount the drive read-only immediately. A single bit flip makes the whole file unusable and there isn't anything we can do to restore it.

I can only recommend you keep your backups in a safer location and also keep multiple versions, not just one. USB thumb drives are not a good backup option. Especially the cheap ones are known for losing or corrupting data willy-nilly. Cloud drives are often a good choice, as they not only offer you offsite backups, but also keep a version history should something go wrong.
